client/src/components/ListDetailDialog/ListDetailDialog.tsx

```diff
@@ -0,0 +1,307 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  FormHelperText,
+  Grid,
+  IconButton,
+  MenuItem,
+  Select,
+  Slide,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import { useEffect, useState } from 'react';
+import useStyles from './useStyles';
+import CloseIcon from '@material-ui/icons/Close';
+import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
+import VisibilityIcon from '@material-ui/icons/Visibility';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { ListInterface } from '../../interface/List';
+import ProductCard from '../ProductCard/ProductCard';
+import { createNewProduct } from '../../helpers/APICalls/product';
+import { ProductInterface } from '../../interface/Product';
+import { validateUrl } from '../../helpers/validateUrl';
+
+interface Props {
+  open: boolean;
+  //   userId: string;
+  setLists: React.Dispatch<React.SetStateAction<ListInterface[]>>;
+  list: ListInterface;
+  lists: ListInterface[];
+  handleClose: () => void;
+}
+
+export default function ListDetailDialog({ open, setLists, list, lists, handleClose }: Props): JSX.Element {
+  const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
+  const [showProducts, setShowProducts] = useState(true);
+  const [showAddNewItem, setShowAddNewItem] = useState(false);
+  const [listState, setListState] = useState<ListInterface>(list);
+  const [newItemUrl, setNewItemUrl] = useState('');
+  const [urlError, setUrlError] = useState<string>('');
+  const [selectedList, setSelectedList] = useState(list._id);
+  const [newProductImageUrl, setNewProductImageUrl] = useState(
+    'https://images-na.ssl-images-amazon.com/images/I/51RRK1PHUOL._AC_SX425_.jpg',
+  );
+  const [newProductName, setNewProductName] = useState('Microsoft Sculpt Ergonomic Keyboard for Business (5KV-00001)');
+  const [newProdutPrice, setNewProdutPrice] = useState('$79.83');
```

Comment:
Should these defaults be empty strings?

Reply:
yep, you're right. I forgot to delete these after testing